Codeinwp/wp-maintenance-mode
# Gutenberg Compatibility [wip]

## Summary

It makes WP Maintenance compatible with Otter Blocks by changing how maintenance pages are built.
A Maintenance Page now uses Gutenberg and templates that use Otter can be imported from the settings page.

## Test Instructions

For new users:
- After activating, an onboarding wizard will appear.
- The wizard has 3 steps: one where the user imports one template from 2 available, one where the user can subscribe to a newsletter, and the last one where they can either go to the maintenance page or the settings page.
- When importing a template from the wizard, a new page named "Maintenance Page" should be created and it is set as the page the entire site will redirect when in maintenance mode.

For old users:
- After updating, a notice should appear where we inform them that some changes are made to build maintenance pages with Gutenberg. They can migrate to the new way by clicking the `Migrate` button from the notice. This notice is undismissible.
- After the migration, changes should be visible in the design tab of the settings page where the user can now select a page as a maintenance page and can import a template, and in the modules tab where we removed the modules that can be replaced with Otter blocks.
- Also, a notice to rollback should be visible and it should disappear after one week if not dismissed.
 
---

- The Maintenance Page will be private when maintenance mode is off. When the maintenance mode is on, the page becomes the front page of the site and every site URL will redirect to this one.

- When the maintenance page gets deleted or drafted while the maintenance mode is on, notices should appear to let the user know some actions are required.

- The bot will be updated for everyone and will be a chat bubble on the bottom right of the screen

- If the Maintenance page has content inside and the user is trying to import a template, a confirmation prompt should inform the user that the content will be replaced. The import will take place only after clicking `Continue`

---

Those who use versions below WP 5.8 can't use Otter so they can't import any template. The wizard will not appear for them and in the design tab, they can only see the templates and a notice saying they need WP >= 5.8.

Closes https://github.com/Codeinwp/otter-blocks/issues/935.
